File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool check(long long num) {
    if(num == 1) return false;
    
    for(long long idx = 2; idx * idx <= num; idx++) {
        if(num % idx == 0) {
            return false;
        }
    }
    
    return true;
}

int solution(int n, int k) {
    int answer = 0;
    
    vector<int> v;
    while(n > 0) {
        v.push_back(n % k);
        n /= k;
    }
    reverse(v.begin(), v.end());
    
    vector<long long> num;
    long long tmp = 0;
    for(int a: v){
        if(a == 0) {
            if(tmp != 0) {
                num.push_back(tmp);
                tmp = 0;
            }
        }
        else {
            tmp = tmp * 10 + a;
        }
    }
    if(tmp != 0) num.push_back(tmp);
    
    for(long long a: num) {
        if(check(a)) answer++;
    }  
    
    return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def convert(n: int, k: int) -> str:
    ret = ""
    
    while n:
        ret = str(n % k) + ret
        n //= k
        
    return ret

def isPrime(n: int) -> bool:
    if n == 1:
         return False
    
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

def solution(n, k):
    answer = 0
    
    n_by_k = convert(n, k)
    
    nums = n_by_k.split("0")
    for num in nums:
        if num == '':
            continue
            
        if isPrime(int(num)):
            answer += 1
    
    return answer